packages/core/src/external/analytics/posthog.ts

```diff
@@ -34,6 +34,18 @@ export const analytics = (params: EventMessageV1, postApiKey?: string) => {
   };
 
   posthog.capture(params);
+
+  return posthog;
+};
+
+export const analyticsAsync = async (params: EventMessageV1, postApiKey?: string) => {
```

Comment:
Nit: can we use a regular function declaration and not an arrow function declaration as per Eng Best Practices Doc?

Reply:
Can we also name this function to reflect its to be used in lambdas, so something like 

analyticsLambda

Reply:
Updated the function type and i think analyticsAsync is more appropriate because there could be other applications for this function